sidestep_engine/data/preprocess.py

```
@@ -103,6 +102,103 @@ def _write_preprocess_meta(
+        mapped = VARIANT_DIR_MAP.get(variant, variant)
+        ckpt_root = Path(checkpoint_dir) / mapped
+        # Try loading just the VAE state dict keys we need
+        import glob
+        safetensor_files = list(ckpt_root.glob("*.safetensors"))
+        if safetensor_files:
+            try:
+                from safetensors.torch import load_file
+                for sf in safetensor_files:
+                    st = load_file(str(sf))
+                    for key in st:
+                        if "decoder.conv1.weight_v" in key or "vae" in key.lower() and "conv1.weight_v" in key:
```

Ruff (0.15.7)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)